vpr/src/route/route_common.cpp

```diff
@@ -401,38 +411,73 @@ void free_route_structs() {
     }
 }
 
-void alloc_and_load_rr_node_route_structs() {
-    /* Allocates some extra information about each rr_node that is used only   *
-     * during routing.                                                         */
+void alloc_and_load_rr_node_route_structs(const t_router_opts& router_opts) {
+    // Allocates some extra information about each rr_node that is used only during routing.
 
     auto& route_ctx = g_vpr_ctx.mutable_routing();
-    auto& device_ctx = g_vpr_ctx.device();
+    const auto& device_ctx = g_vpr_ctx.device();
 
     route_ctx.rr_node_route_inf.resize(device_ctx.rr_graph.num_nodes());
     route_ctx.non_configurable_bitset.resize(device_ctx.rr_graph.num_nodes());
     route_ctx.non_configurable_bitset.fill(false);
 
-    reset_rr_node_route_structs();
+    reset_rr_node_route_structs(router_opts);
 
     for (auto i : device_ctx.rr_node_to_non_config_node_set) {
         route_ctx.non_configurable_bitset.set(i.first, true);
     }
 }
 
-void reset_rr_node_route_structs() {
-    /* Resets some extra information about each rr_node that is used only   *
-     * during routing.                                                         */
+static float comp_initial_acc_cost(RRNodeId node_id,
```

Comment:
Explain the reasoning / intuition behind this.  "Usually acc_cost starts at 1 for all rr_nodes, but we ... to push the router to ..."